src/Tickets/Blocks/Ticket/app/editor/container-content/type/container.js

```diff
@@ -0,0 +1,35 @@
+/**
+ * External dependencies
+ */
+import { connect } from 'react-redux';
+import { compose } from 'redux';
+
+/**
+ * WordPress dependencies
+ */
+import { select } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import Template from './template';
+import { withStore } from '@moderntribe/common/hoc';
+import { selectors } from '@moderntribe/tickets/data/blocks/ticket';
+
+const mapStateToProps = ( state, ownProps ) => {
+	const postType = select( 'core/editor' ).getCurrentPostType();
+	const ticketDetails = selectors.getTicketDetails( state, ownProps );
+
+	return ( {
+		postType: postType.replace( 'tribe_events', 'event' ),
+		type: ticketDetails.type,
+		seriesTitle: 'TODO this',
```

Comment:
None of these components should contain hard-coded information about the post or the ticket types.
All the information required should come from the backend, likely from the `ticketDetails`.

```js
return ( {
  postType,
  type: ticketDetails.type
  iconUrl: ticketDetails.typeIconUrl
  description: ticketDetails.typeDescription
} );

That data is already present in the backend [to fuel the same kind of label in the Classic Editor](https://github.com/the-events-calendar/event-tickets/blob/66566a96897aa85f9272e9f9d8f6cb3612db11e4/src/Tickets/Flexible_Tickets/Metabox.php#L177-L188); it should be pulled from there.

The data should come from [the `fetchTicket` selector](https://github.com/the-events-calendar/event-tickets/blob/692a6599a78621ca0623b0fb05bd49eb4162136d/src/modules/data/blocks/ticket/sagas.js#L380-L400) and be just available on the ticket.

Knowing the data should come from the backend, likely from the `ticketDetails`, let me know what values you will need to make this component completely independent of the post and ticket type and be completely powered by information coming from the backend.

The only values that should live in this components are the ones for the `default` type. Those would be required to hydrate an edit form for a new ticket.

Reply:
Thanks for the observation. The only missing property from the backed I think it would be the typeName, that is [here](https://github.com/the-events-calendar/event-tickets/pull/2833/files#diff-ffa65215cfabf674e4fa1a35dd47dfa8aaabf122a04170da19f00919c2426126R29)